### DISCUSSION TITLE
Rounded border for hovered row in the file list

### DISCUSSION THREAD

**Discussion by MinaMatta98** [sxyazi/yazi#426] [1. Q&A] [answered]
### Will you be willing to contribute this feature?

- [ ] Yes, I'll give it a shot

### Describe the solution you'd like

Hello,

Would you consider supporting rounded borders around current and future selections, as such:

![image](https://github.com/sxyazi/yazi/assets/81748812/3a8da170-c41a-4655-ba9e-6d275014cfed)

This may really improve the feel of the tui.

**Comment by sxyazi** [comment]
Hmm, how did you make it rounded? 😂

**Comment by MinaMatta98** [comment]
Oh, haha,

I used a screenshot and colored in the edges. HAHAHA

**Comment by sxyazi** [comment, marked as answer]
https://github.com/sxyazi/yazi/assets/17523360/fb70ff66-51c4-42b0-9159-327e219a32f2

Yazi allows you to rewrite most of the UI through UI plugins, and you can achieve this by extending the [`linemode`](https://yazi-rs.github.io/docs/configuration/yazi#:~:text=and%20sort_reverse%20only-,linemode,-%3A%20Line%20mode):

```lua
function Folder:linemode(area)
  local lines = {}
  for _, f in ipairs(self:by_kind(self.CURRENT).window) do
    if f:is_hovered() then
      -- You may want to change the "foreground color" here to adapt to your theme
      lines[#lines + 1] = ui.Line { ui.Span(""):bg("reset"):fg("blue") }
      break
    else
      lines[#lines + 1] = ui.Line {}
    end
  end
  return ui.Paragraph(area, lines):align(ui.Alignment.RIGHT)
end
```

Save those lines as a file, for example `~/.config/yazi/ui.lua`, and include it in the `yazi.toml`:

```toml
[plugins]
preload = [
  "~/.config/yazi/ui.lua"
]
```

This is just an example, please read the documentation about `linemode`: https://yazi-rs.github.io/docs/configuration/yazi#:~:text=and%20sort_reverse%20only-,linemode,-%3A%20Line%20mode

**Reply by MinaMatta98** [reply]
Does using lua lead to a significant performance decrease?

**Reply by sxyazi** [reply]
No, all layouts are implemented in Rust, Lua merely provides a representation layer for layout configuration.

Also, all state data is native to Rust and provided to Lua in a read-only shared manner, thus avoiding unnecessary memory allocation — they're just pointers, so they're inexpensive.

**Reply by MinaMatta98** [reply]
Awesome.

Did you have an API reference for the lua?

Or is https://github.com/sxyazi/yazi/tree/main/yazi-plugin/preset sufficient?

**Reply by sxyazi** [reply]
Nope, this isn't sufficient, but it's all we have for now. I'll find time to add more documents.

**Reply by MinaMatta98** [reply]
Sorry, could you also detail how to apply the same to the parent?

**Reply by sxyazi** [reply]
ill find time to try implementing rounded edges for other directories
